### PR TITLE
refactor: put `dataframe` & `query` into `greptime` module

### PR DESCRIPTION
**Pull request by discord9**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?

Put `dataframe` and `query` into `greptime` module
notice this currently only works for PyO3 backend, because RustPython doesn't support modifying modules after creation of interpreter(unless init interpreter per call of script, which is way too costy)

Please explain IN DETAIL what the changes are in this PR and why they are needed:

- Summarize your change (**mandatory**)
do this in python script(Only in pyo3) to get access to dataframe api and query in sql
`from greptime import dataframe, query`
- How does this PR work? Need a brief introduction for the changed logic (optional)
added `dataframe` & `query`  into `greptime` module in pyo3 every time coprocessor is called with a RecordBatch and within a `QueryEngine`, and use it like:
`query().sql(...)` and `dataframe().collect()...`
- Describe clearly one logical change and avoid lazy messages (optional)
- Describe any limitations of the current code (optional)
This works in pyo3 module by dynamically insert both var into module everytime, and in rustpython backend is using hidden variable `__dataframe__` and `__query__`

## Checklist

- [ ]  I have written the necessary rustdoc comments.
- [ ]  I have added the necessary unit tests and integration tests.

### CODE REVIEW

**Review by killme2008 — APPROVED** [review]
Good job! LGTM.

**Comment by killme2008** [comment]
@waynexia  PTAL

File: src/script/src/python/rspython/test.rs [hunk 1]
```diff
@@ -287,7 +287,7 @@ def a(cpu, mem):
     abc = vector([v[0] > v[1] for v in zip(cpu, mem)])
     fed = cpu.filter(abc)
     ref = log2(fed/prev(fed))
-    return (0.5 < cpu) & ~( cpu >= 0.75)
+    return cpu[(cpu > 0.5) & ~( cpu >= 0.75)]
```
Comment (waynexia) [inline src/script/src/python/rspython/test.rs:290]: nit:
```suggestion
    return cpu[(cpu > 0.5) & ~(cpu >= 0.75)]
```

File: src/script/src/python/rspython/builtins.rs [hunk 2]
```diff
@@ -306,13 +306,38 @@ pub(crate) mod greptime_builtin {
...
+        let df = vm.current_globals().get_item("__dataframe__", vm)?;
+        let df = df
+            .payload::<PyDataFrame>()
+            .ok_or_else(|| vm.new_type_error(format!("object {:?} is not a DataFrame", df)))?;
+        let df = df.clone();
+        Ok(df)
+    }
+
+    /// get `__query__` from globals and return it
+    /// TODO(discord9): this is a terrible hack, we should find a better way to get `__dataframe__`
```
Comment (waynexia) [inline src/script/src/python/rspython/builtins.rs:330]: nit:

```suggestion
    /// TODO(discord9): this is a terrible hack, we should find a better way to get `__query__`
```

By the way, for query engine I think it's ok to store it globally. Like this, TLS or static variable.

**Review by waynexia — APPROVED** [review]
🚀 🚀 🚀

**Comment by discord9** [comment]
Still updating relevant docs: https://github.com/GreptimeTeam/docs/pull/252

**Pull request merged**